bench/syclblas_benchmark.cpp

```diff
@@ -0,0 +1,396 @@
+  }
+
+  template <typename ValueType>
+  static vector_view<ValueType, cl::sycl::buffer<ValueType>> mkvview(
+      cl::sycl::buffer<ValueType, 1> &buf) {
+    return vector_view<ValueType, cl::sycl::buffer<ValueType>>(buf);
+  }
+
+  BENCHMARK_FUNCTION(scal_bench) {
+    UNPACK_PARAM;
+    ScalarT *v1 = new_data<ScalarT>(size);
+    ScalarT alpha(2.4367453465);
+    double flops;
+    {
+      auto buf1 = mkbuffer<ScalarT>(v1, size);
+      auto vvw1 = mkvview(buf1);
+      flops = benchmark<>::measure(no_reps, size * 1, [&]() {
+        _scal(ex, size, alpha, vvw1, 1);
+        q.wait_and_throw();
+      });
+    }
+    release_data(v1);
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(axpy_bench) {
+    UNPACK_PARAM;
+    ScalarT *v1 = new_data<ScalarT>(size);
+    ScalarT *v2 = new_data<ScalarT>(size);
+    ScalarT alpha(2.4367453465);
+    double flops;
+    {
+      auto buf1 = mkbuffer<ScalarT>(v1, size);
+      auto buf2 = mkbuffer<ScalarT>(v2, size);
+      auto vvw1 = mkvview(buf1);
+      auto vvw2 = mkvview(buf2);
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        _axpy(ex, size, alpha, vvw1, 1, vvw2, 1);
+        q.wait_and_throw();
+      });
+    }
+    release_data(v1);
+    release_data(v2);
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(asum_bench) {
+    UNPACK_PARAM;
+    ScalarT *v1 = new_data<ScalarT>(size);
+    ScalarT vr;
+    double flops;
+    {
+      auto buf1 = mkbuffer<ScalarT>(v1, size);
+      auto bufR = mkbuffer<ScalarT>(&vr, 1);
+      auto vvw1 = mkvview(buf1);
+      auto vvwR = mkvview(bufR);
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        _asum(ex, size, vvw1, 1, vvwR);
+        q.wait_and_throw();
+      });
+    }
+    release_data(v1);
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(nrm2_bench) {
+    UNPACK_PARAM;
+    ScalarT *v1 = new_data<ScalarT>(size);
+    ScalarT vr;
+    double flops;
+    {
+      auto buf1 = mkbuffer<ScalarT>(v1, size);
+      auto bufR = mkbuffer<ScalarT>(&vr, 1);
+      auto vvw1 = mkvview(buf1);
+      auto vvwR = mkvview(bufR);
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        _nrm2(ex, size, vvw1, 1, vvwR);
+        q.wait_and_throw();
+      });
+    }
+    release_data(v1);
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(dot_bench) {
+    UNPACK_PARAM;
+    ScalarT *v1 = new_data<ScalarT>(size);
+    ScalarT *v2 = new_data<ScalarT>(size);
+    ScalarT vr;
+    double flops;
+    {
+      auto buf1 = mkbuffer<ScalarT>(v1, size);
+      auto buf2 = mkbuffer<ScalarT>(v2, size);
+      auto bufR = mkbuffer<ScalarT>(&vr, 1);
+      auto vvw1 = mkvview(buf1);
+      auto vvw2 = mkvview(buf2);
+      auto vvwR = mkvview(bufR);
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        _dot(ex, size, vvw1, 1, vvw2, 1, vvwR);
+        q.wait_and_throw();
+      });
+    }
+    release_data(v1);
+    release_data(v2);
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(iamax_bench) {
+    UNPACK_PARAM;
```

Comment:
Yes, but I wanted to make the same template arguments, return value and parameters which would be defined in benchmarks.hpp.